src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.MovieDao;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity", e);
```

Comment:
```suggestion
            throw new RuntimeException("Can't insert movie entity", e);
```
here we should throw our custom exception